doc/example/limitations_const_parameter_warning.cpp

```diff
@@ -48,3 +47,10 @@ namespace limitations_const_parameter_warning_solution
     };
 //]
 }
+
+BOOST_AUTO_TEST_CASE(check_method_stub_is_called)
+{
+    mock_base b;
+    MOCK_EXPECT(b.method).once().with(1);
+    static_cast<base*>(&b)->method(1);
```

Comment:
Does this need a `static_cast`? Also it could probably be a reference?

Reply:
Yes it does. You usually pass the mocked object as a pointer to base into a method to test. Without the cast the virtual dispatch won't happen as the derived (mock) object instance is used. But for real world testing we need to test it under real world conditions. Does this make sense?

Reply:
> You usually pass the mocked object as a pointer to base into a method to test

Do you think people still pass naked pointers around much these days? More than references?
(genuine question, I haven’t done much C++ in the past 2 years)

Reply:
Yes usually this would be a `unique_ptr` or similar or a reference. Most likely it would be some kind of pointer though, so that's why I'm casting to a pointer. I'll change to a lambda to avoid the explicit cast, ok?

Reply:
What would it look like with a lambda?

Reply:
Changed:
```
    // Example user code taking a base* (or smart pointer variant)
    auto callMethod = [](base* bPtr){ bPtr->method(1); };
    callMethod(&b);
```